internal/transport/transport_test.go

```diff
@@ -461,40 +461,151 @@ func setUpWithOptions(t *testing.T, port int, sc *ServerConfig, ht hType, copts
 	return server, ct.(*http2Client), cancel
 }
 
-func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.Conn) (*http2Client, func()) {
-	lis, err := net.Listen("tcp", "localhost:0")
+// A server which can be switched from responsive to non-responsive by writing a toggle boolean to the respModeCh channel
+// if a respModeCh channel is created.
```

Comment:
```suggestion
// pingModeServer is a custom server which can switch from responding to pings
// to ignoring them and vice versa using signals sent on a boolean chan
// respModeCh. If respModeCh is nil, the server would work as a no ping server.
```

Reply:
If you follow my suggestion below, then the usage of a boolean chan to control the PingMode can be opaque to the user. Instead mention how they can just use `setPingMode` to alter the PingMode of the server.